controller/controller.go

```diff
@@ -31,6 +31,13 @@ const (
 	loggerKeyObject     = "object"
 	loggerKeyResource   = "resource"
 	loggerKeyVersion    = "version"
+
+	// removedFinalizersCacheSize should be bigger than number of
+	// reconciled objects to handle all deletions at the same time. Even if
+	// it is too small in the worst case scenario we will get deletion
+	// event for already deleted object. 1000 seems to be plenty. It's hard
+	// to imagine deletion of 1000 reconciled objects at a time.
+	removedFinalizersCacheSize = 1000
```

Comment:
Without checking the code and understanding the implementation, my assumption would be that this is a threshold over time. So there have to be 1001 deletion events concurrently across time in order to cause multiple delete event duplications. So in case there are once 1001 delete events at peak, the next reconciliation loops will resolve the situation, because the duplicated events will disappear, because they are delete events and vanish once finished.

Reply:
This is no longer true. With the new cache implementation we should handle any number of events.